hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -159,10 +161,15 @@ public void updateSnapshotsFrom(FunctionalityThrottling throttling) {
 		if (n == 0) {
 			usageSnapshots = NO_SNAPSHOTS;
 		} else {
-			usageSnapshots = new DeterministicThrottle.UsageSnapshot[n];
 			for (int i = 0; i < n; i++) {
-				usageSnapshots[i] = activeThrottles.get(i).usageSnapshot();
+				var usageSnapshot = activeThrottles.get(i).usageSnapshot();
```

Comment:
This call to `activeThrottles.get(i).usageSnapshot()` is exactly the "deep copy" we want to eliminate (since the [`DeterministicThrottle.usageSnapshot()` method](https://github.com/hashgraph/hedera-services/blob/master/hapi-utils/src/main/java/com/hedera/services/throttles/DeterministicThrottle.java#L140) creates a **new** `UsageSnapshot`.

Reply:
Thank you for pointing it out. Will change the implementation here.